packages/multi-select-combo-box/src/vaadin-multi-select-combo-box-internal.js

```diff
@@ -326,6 +326,21 @@ class MultiSelectComboBoxInternal extends ComboBoxDataProviderMixin(ComboBoxMixi
 
     return super._shouldLoadPage(page);
   }
+
+  /**
+   * Override method inherited from the combo-box
+   * to not clear the data provider cache when read-only.
+   *
+   * @protected
+   * @override
+   */
+  clearCache() {
+    if (this.readonly) {
+      return;
```

Comment:
What happens if the component is `readOnly` and the items data changes? Or if the component is initially `readOnly` and some data and selection is assigned to it?

Reply:
There are special computed properties that substitute `filteredItems` and `items` of the internal combo-box when read-only (If I understand your question):

https://github.com/vaadin/web-components/blob/1ae03e8d854cb3ccd85fd71ae51df10d2e997e09/packages/multi-select-combo-box/src/vaadin-multi-select-combo-box.js#L1115-L1123

That being said, it seems there is a case that I overlooked. We apparently need to call `cleanCache()` on read-only attribute removal.

Reply:
> What happens if the component is readOnly and the items data changes? Or if the component is initially readOnly and some data and selection is assigned to it?

Basically, if you assign new items while the component is `readOnly`, they won't be passed to the internal combo-box until `readOnly` is off.

Reply:
> That being said, it seems there is a case though that I overlooked. We apparently need to call cleanCache() on read-only attribute removal.

Addressed in https://github.com/vaadin/web-components/pull/4598/commits/c98bed51520419775e5ba1fe49580d6e6480dc69

Reply:
My concern was about updating the data items (like the item labels) dynamically.

Tested the imaginary scenario, and the behavior seems somewhat off on the master also:

```html
<vaadin-multi-select-combo-box
  label="Element"
  item-id-path="value"
  item-label-path="label"
  readonly
></vaadin-multi-select-combo-box>

<vaadin-multi-select-combo-box
  label="Element"
  item-id-path="value"
  item-label-path="label"
></vaadin-multi-select-combo-box>

<script>
  const comboBoxes = document.querySelectorAll('vaadin-multi-select-combo-box');

  [...comboBoxes].forEach(comboBox => {
    // Set the selected item for a combo box that identifies the items by "value" property
    comboBox.selectedItems = [
      {
        label: 'Hydrogen',
        value: 'H'
      },
    ];

    // Update the data (the updated label should be used in the chip as well as in the dropdown)
    comboBox.items = [
      {
        label: 'Updated Hydrogen',
        value: 'H'
      }
    ];

  });
</script>

```

https://user-images.githubusercontent.com/1222264/190360783-9e179308-0121-47cf-9f61-0a0ea8bac418.mp4

Issues discovered:
- Initially, both combos show the old label in the chip
- The readOnly combo shows the old label in dropdown also
- When the updated item is selected, the combo shows "1" as the chip value

Reply:
Seems that the dropdown shows an outdated label also on when a dataProvider & clearCache is used, but it's not a regression from this PR, so not blocking

Reply:
I guess it is because `multi-select-combo-box` allows multiple items to be selected so this part doesn't work for it:

https://github.com/vaadin/web-components/blob/183b85765f252c8b664a6c4835263018a4ce2d46/packages/combo-box/src/vaadin-combo-box-mixin.js#L1136-L1139